ksql-parser/src/main/java/io/confluent/ksql/schema/ksql/DefaultSqlValueCoercer.java

```diff
@@ -62,6 +77,43 @@
     return optional(result);
   }
 
+  private static <T> Optional<T> coerceArray(final Object value, final SqlArray targetType) {
+    if (!(value instanceof List<?>)) {
+      return Optional.empty();
+    }
+
+    final List<?> list = (List<?>) value;
+    final ImmutableList.Builder<Object> coerced = ImmutableList.builder();
+    for (final Object el : list) {
+      final Optional<Object> coercedEl = doCoerce(el, targetType.getItemType());
+      if (!coercedEl.isPresent()) {
+        return Optional.empty();
+      }
+      coerced.add(coercedEl.get());
+    }
+
+    return optional(coerced.build());
+  }
+
+  private static <T> Optional<T> coerceMap(final Object value, final SqlMap targetType) {
```

Comment:
As above but for Map

Reply:
Same comment as above - I think it makes more sense not to type it as a Map, but I removed the type parameter.